From a6750e8af7803986945ccfd713f2617a32a0851d Mon Sep 17 00:00:00 2001
From: Stafford Horne <shorne@gmail.com>
Date: Wed, 6 Oct 2021 08:31:12 +0900
Subject: [PATCH 12/14] libio: Update tst-wfile-sync to not depend on stdin

The test expects stdin to be a file which is not the case when running
tests over ssh where stdin is piped in.

The test fails with:
  error: xlseek.c:27: lseek64 (0, 0, 1): Illegal seek

Update the test to create a temporary file and use that to perform the
test.

Reviewed-by: Adhemerval Zanella  <adhemerval.zanella@linaro.org>

(cherry-picked from commit 8faa1e04493f23b16f473d21a3a5bc49b781ed2a)
---
 libio/tst-wfile-sync.c     | 27 ++++++++++++++++++++++++---
 libio/tst-wfile-sync.input |  1 -
 2 files changed, 24 insertions(+), 4 deletions(-)
 delete mode 100644 libio/tst-wfile-sync.input

diff --git a/libio/tst-wfile-sync.c b/libio/tst-wfile-sync.c
index 5194164a2c..c19d010223 100644
--- a/libio/tst-wfile-sync.c
+++ b/libio/tst-wfile-sync.c
@@ -16,22 +16,43 @@
    License along with the GNU C Library; if not, see
    <https://www.gnu.org/licenses/>.  */
 
+#include <fcntl.h>
 #include <locale.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include <wchar.h>
 #include <support/check.h>
+#include <support/xstdio.h>
 #include <support/xunistd.h>
+#include <support/temp_file.h>
+
+static const char test_data[] = "This is a test of _IO_wfile_sync.";
 
 static int
 do_test (void)
 {
+  static char *infile;
+  int infd;
+  FILE *infp;
+
+  infd = create_temp_file ("tst-wfile-sync-in-", &infile);
+  xwrite (infd, test_data, strlen (test_data));
+  xclose (infd);
+
+  infd = xopen (infile, O_RDONLY, 0);
+  infp = fdopen (infd, "r");
+
   TEST_VERIFY_EXIT (setlocale (LC_ALL, "de_DE.UTF-8") != NULL);
   /* Fill the stdio buffer and advance the read pointer.  */
-  TEST_VERIFY_EXIT (fgetwc (stdin) != WEOF);
+  TEST_VERIFY_EXIT (fgetwc (infp) != WEOF);
   /* This calls _IO_wfile_sync, it should not crash.  */
-  TEST_VERIFY_EXIT (setvbuf (stdin, NULL, _IONBF, 0) == 0);
+  TEST_VERIFY_EXIT (setvbuf (infp, NULL, _IONBF, 0) == 0);
   /* Verify that the external file offset has been synchronized.  */
-  TEST_COMPARE (xlseek (0, 0, SEEK_CUR), 1);
+  TEST_COMPARE (xlseek (infd, 0, SEEK_CUR), 1);
+
+  fclose (infp);
+  free (infile);
 
   return 0;
 }
diff --git a/libio/tst-wfile-sync.input b/libio/tst-wfile-sync.input
deleted file mode 100644
index 12d0958f7a..0000000000
--- a/libio/tst-wfile-sync.input
+++ /dev/null
@@ -1 +0,0 @@
-This is a test of _IO_wfile_sync.
-- 
2.34.1

